### ISSUE TITLE
Modify update URL opening method

### ISSUE THREAD

**Issue by LiroyvH** [MediaGrabber/iOS-Workflow#2]
Hi,

When the JSON output shows there's an update, it'll grab update_url and open it in Safari (Open URL). Please consider using x-callback-URL instead. (Open x-callback-URL)

This ensures that users who use another browser than Safari as default on their iPhone/iPad will be able to grab the updated workflow. Otherwise, you'll get an error that Safari cannot be opened and thus the Workflow ends itself without update and without grabbing the video as the update is forced. And believe me, removing Safari and adding x-callback-URL in manually is a loooonnngggg way to scroll up. ;) Only to be deleted by the update again, so on next update you hit the same issue.
Changing it to x-callback-URL solves this problem, as Workflow-app actually opens it directly as it recognises it. It doesn't even seem to open the browser anymore, the app itself handles it. That's also faster as it prevents the opening of another app to get the update. 

By the way, if ever a copy of the server-side script with youtubedl to generate all the JSON stuff can be shared would be awesome. On some locations the call to the server is blocked by firewall on the wifi vlan, but it could call a local server with fewer restrictions to generate it and thus obtain it.
Of course this would require manually altering the workflow as well to call that, but that's less of a problem than not being able to connect at all.

Thanks.

**Comment by sayem314** [comment]
But wouldn’t that drop support for workflow users? Does workflow have x-callback opener?

And isn’t safari always the default browser for shortcuts or any other official apple apps?

**Comment by LiroyvH** [comment]
@sayem314 

"But wouldn’t that drop support for workflow users? Does workflow have x-callback opener?"
I don't know. Workflow got converted to Shortcuts for me so I can't test it with users of the old app. However, if you have Workflow installed it's very simple to test - all you need to do is modify Safari to x-callback-URL and on next update: see what it does. :)

But all the same, if Workflow doesn't have a registered x-callback-URL function (such as recognising .wflow files from download), then the phone will default to the next best thing: Safari or another browser. And then it will be able to open/share to Workflow.
As such, there should be no adverse affects to changing it. All it really does is first look if there's a compatible application to handle the URL. If there isn't: it'll pass it to the default browser, which is either Safari or whatever people have set. So in the best case scenario: you skip a step.
In the worst case scenario: it should do the exact same thing as before and users shouldn't notice any difference. But as always, testing it is always the best option - but I can't test it myself. 
 

"And isn’t safari always the default browser for shortcuts or any other official apple apps?"

No not anymore. :) First of it depends on what the app registered itself for, there's multiple schemes apps can register themselves for with x-callback which bypasses Safari pretty much completely - like how Wikipedia works. (Which is also for example why after paying somewhere, your browser can ask you if you wish to load the result in to that application; it recognises the registered callback and acts on it.)  Secondly, you can set another default browser in iOS by getting one that registers itself to handle URL's and then simply disable Safari. This will cause your other browser to become the default to handle everything. :)

**Comment by sayem314** [comment]
Thanks. Will add this on next version.

**Comment by LiroyvH** [comment]
@sayem314 Thanks, that sounds good! :)

Also, and let me know if you'd want a new issue opened for this, it'd be very nice to have an option "Create GIF". I added it in now as option (Create GIF, 0.1 sec per frame, Save to Camera roll), but that will of course be erased upon next update. For example if you want to download a GIF from Reddit, it will get you an mp4 instead. (That's due to how Reddit works). So you'd have to download it to your phone, then use another Shortcut or an app like ImgPlayPro to convert it back to GIF. Much easier if the function is built-in to MediaGrabber. 

Thank you for your consideration! =)

**Comment by sayem314** [comment]
Media Grabber is for videos and audio only and I don’t want it to have any kind of functionality for images or gif.

You should make a shortcuts instead of modifying MG on every update for persistent changes so everytime you install new version of MG you can plug your shortcuts easily.

On every new updates of MG change save to camera roll action to run shortcuts action with your shortcuts and handle stuff from there.
